java/code/src/com/suse/manager/xmlrpc/iss/IssHandler.java

```diff
@@ -0,0 +1,291 @@
+ */
+public class IssHandler extends BaseHandler {
+
+    private static final Logger LOGGER = LogManager.getLogger(IssHandler.class);
+
+    private final HubManager hubManager;
+
+    /**
+     * Default constructor
+     */
+    public IssHandler() {
+        this(new HubManager());
+    }
+
+    /**
+     * Builds a handler with the specified dependencies
+     * @param hubManagerIn the hub manager
+     */
+    public IssHandler(HubManager hubManagerIn) {
+        this.hubManager = hubManagerIn;
+    }
+
+    /**
+     * Generate a new access token for ISS for accessing this system
+     * @param loggedInUser the user logged in. It must have the sat admin role.
+     * @param fqdn the FQDN of the peripheral/hub that will be using this access token
+     *
+     * @return the serialized form of the token
+     *
+     * @apidoc.doc Generate a new access token for ISS for accessing this system
+     * @apidoc.param #session_key()
+     * @apidoc.param #param_desc("string", "fqdn", "FQDN of the peripheral/hub that will be using this access token")
+     * @apidoc.returntype #param("string", "The serialized form of the token")
+     */
+    public String generateAccessToken(User loggedInUser, String fqdn) {
+        ensureSatAdmin(loggedInUser);
+
+        if (StringUtils.isEmpty(fqdn)) {
+            throw new InvalidParameterException("No FQDN specified");
+        }
+
+        try {
+            return hubManager.issueAccessToken(fqdn);
+        }
+        catch (TokenException ex) {
+            LOGGER.error("Unable to issue a token for {}", fqdn, ex);
+            throw new TokenCreationException();
+        }
+        catch (ConstraintViolationException ex) {
+            LOGGER.error("Unable to issue a token, it already exists for {}", fqdn, ex);
+            throw new TokenAlreadyExistsException();
+        }
+    }
+
+    /**
+     * Store a third party access token for ISS
+     * @param loggedInUser the user logged in. It must have the sat admin role.
+     * @param fqdn the FQDN of the peripheral/hub that generated this access token
+     * @param token the access token
+     * @return 1 on success, exception otherwise
+
+     * @apidoc.doc Generate a new access token for ISS for accessing this system
+     * @apidoc.param #session_key()
+     * @apidoc.param #param_desc("string", "fqdn", "the FQDN of the peripheral/hub that generated this access token")
+     * @apidoc.param #param_desc("string", "token", "the access token")
+     * @apidoc.returntype #return_int_success()
+     */
+    public int storeAccessToken(User loggedInUser, String fqdn, String token) {
+        ensureSatAdmin(loggedInUser);
+
+        if (StringUtils.isEmpty(fqdn)) {
+            throw new InvalidParameterException("No FQDN specified");
+        }
+
+        if (StringUtils.isEmpty(token)) {
+            throw new InvalidParameterException("No token specified");
+        }
+
+        try {
+            hubManager.storeAccessToken(fqdn, token);
```

Comment:
I wonder if this method should also get a loggedInUser parameter to check it inside of it.
If this method is also called by UI orther code it might be better to perform the user check also in the manager class.